collect_app/src/main/java/org/odk/collect/android/formentry/QuitFormDialogFragment.java

```diff
@@ -60,6 +57,12 @@ public class QuitFormDialogFragment extends DialogFragment {
     private FormEntryViewModel formEntryViewModel;
     private Listener listener;
 
+    private final FormSaveViewModel.FactoryFactory formSaveViewModelFactoryFactory;
+
+    public QuitFormDialogFragment(FormSaveViewModel.FactoryFactory formSaveViewModelFactoryFactory) {
```

Comment:
I've been going back and forth on if we should be thinking about replacing Dagger in Fragments with Fragment constructors (and `FragmentFactory`) and the direction here is a big push there. I'm thinking we should look at restructuring how we generally do `ViewModel` factories and create one per Activity that can then be passed to Fragment constructors. That would mean that here the constuctor could just be:

```kotlin
class QuitFormDialogFragment(private val viewModelFactory: ViewModelProvider.Factory) {
```

...and then we'd have a single factory for `FormEntryActivity`:

```kotlin
class Factory(...dependencues...) : ViewModelProvider.Factory {
        override fun <T : ViewModel> create(modelClass: Class<T>): T {
            when (modelClass) {
                FormEntryViewModel::class.java -> { ... }
                FormSaveViewModel::class.java -> { ... }
                ...
            }
        }
}
```

As an aside, this approach would also make moving to Hilt (#4940) easier, as we'd have less of a problem with the incompatibility with Fragment.

I'd like to have finished the though in this PR, but reorganizing the view model construction for `FormEntryActivity` is probably not something we'd want to release as part of a hotfix, so I think it makes sense to discuss here briefly, and then create an issue (or I just follow up immediately with a different PR).